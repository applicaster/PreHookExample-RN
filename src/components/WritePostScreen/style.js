import {
  Dimensions,
  StyleSheet,
} from 'react-native';

const window = Dimensions.get('window');

export const styles = StyleSheet.create({
  writePostScreen: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#323232',
  },
  input: {
    height: 205,
    fontSize: 20,
    backgroundColor: '#FFFFFF',
    paddingHorizontal: 10,
    paddingTop: 15,
    width: window.width,
  },
  postBar: {
    flexDirection: 'row',
    justifyContent: 'center',
    height: 65,
    width: window.width,
  },
  postButton: {
    position: 'absolute',
    right: 12,
    top: 32,
  },
  postButtonLabel: {
    fontSize: 18,
    fontWeight: '600',
  },
  writePostLabel: {
    fontWeight: '600',
    fontSize: 18,
    top: 32,
  },
  postOptionsBar: {
    alignItems: 'center',
    flexDirection: 'row',
    height: 45,
    width: window.width,
  },
});
