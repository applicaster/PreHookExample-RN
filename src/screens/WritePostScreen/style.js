import {
  Dimensions,
  StyleSheet,
} from 'react-native';

const window = Dimensions.get('window');
export const BAR_HEIGHT = 45;
export const STATUS_BAR_HEIGHT = 20;

export const styles = StyleSheet.create({
  writePostScreen: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#323232',
  },
  input: {
    fontSize: 20,
    backgroundColor: '#FFFFFF',
    color: '#000000',
    paddingHorizontal: 10,
    paddingTop: 15,
    textAlignVertical: 'top',
    width: window.width,
  },
  postBar: {
    flexDirection: 'row',
    justifyContent: 'center',
    height: BAR_HEIGHT + STATUS_BAR_HEIGHT,
    width: window.width,
  },
  postButton: {
    position: 'absolute',
    right: 12,
    top: 31,
  },
  postButtonLabel: {
    fontSize: 18,
    fontWeight: '600',
  },
  writePostLabel: {
    fontWeight: '600',
    fontSize: 18,
    top: 32,
  },
  postOptionsBar: {
    alignItems: 'center',
    flexDirection: 'row',
    height: BAR_HEIGHT,
    width: window.width,
  },
});
