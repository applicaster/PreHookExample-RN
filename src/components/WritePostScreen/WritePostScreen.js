import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Alert,
  Dimensions,
  Keyboard,
  Platform,
  TextInput,
  Text,
  View,
  TouchableOpacity,
} from 'react-native';
import FeedRNUtils from '@applicaster/feed-rn-utils';
import { styles, BAR_HEIGHT, STATUS_BAR_HEIGHT } from './style';
import CloseButton from '../CloseButton';
import CharacterCounter from '../CharacterCounter';
import PostSwitcher from '../PostSwitcher';

const window = Dimensions.get('window');
class WritePostScreen extends Component {
  constructor(props) {
    super(props);
    const { twitterScreenName } = props;
    this.state = {
      facebookText: '',
      twitterText: (twitterScreenName) ? `@${twitterScreenName} ` : '',
      keyboardHeight: 0,
      socialNetworkSelected: 'twitter',
    };
    
    this.closeModal = this.closeModal.bind(this);
    this.onTextChange = this.onTextChange.bind(this);
    this.onPostPress = this.onPostPress.bind(this);
    this.keyboardWillShow = this.keyboardWillShow.bind(this);
    this.keyboardWillHide = this.keyboardWillHide.bind(this);
    this.toggleNetworkSelected = this.toggleNetworkSelected.bind(this);
  }

  componentWillMount() {
    this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
    this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
  }

  componentWillUnmount() {
    this.keyboardWillShowSub.remove();
    this.keyboardWillHideSub.remove();
  }

  onTextChange(text) {
    const { socialNetworkSelected } = this.state;
    if (socialNetworkSelected === 'facebook') {
      this.setState({ facebookText: text });
    } else if (socialNetworkSelected === 'twitter') {
      this.setState({ twitterText: text });
    }
  }

  onPostPress() {
    const {
      facebookText,
      twitterText,
      socialNetworkSelected,
    } = this.state;

    const {
      okButtonText,
      facebookErrorTitle,
      facebookErrorMessage,
      twitterErrorTitle,
      twitterErrorMessage,
    } = this.props;

    if (socialNetworkSelected === 'twitter') {
      if (!twitterText.length) return;

      FeedRNUtils.postTweet(`${twitterText}`)
        .then(() => this.closeModal())
        .catch(() => {
          Alert.alert(
            twitterErrorTitle,
            twitterErrorMessage,
            [{ text: okButtonText, onPress: this.closeModal }],
            { cancelable: false }
          );
        });
    }

    if (socialNetworkSelected === 'facebook') {
      if (!facebookText.length) return;

      const { facebookPageId } = this.props;
      FeedRNUtils.postFacebook({ postText: facebookText, facebookPageId })
        .then(() => this.closeModal())
        .catch(() => {
          Alert.alert(
            facebookErrorTitle,
            facebookErrorMessage,
            [{ text: okButtonText, onPress: this.closeModal }],
            { cancelable: false }
          );
        });
    }
  }

  keyboardWillShow(event) {
    this.setState({ keyboardHeight: Math.floor(event.endCoordinates.height) });
  }

  keyboardWillHide() {
    this.setState({ keyboardHeight: 0 });
  }

  closeModal() {
    const { toggleModal } = this.props;
    toggleModal({ modal: 'WritePostModal' });
  }
  
  toggleNetworkSelected() {
    const { socialNetworkSelected } = this.state;
    this.setState({ socialNetworkSelected: (socialNetworkSelected === 'twitter') ? 'facebook' : 'twitter' });
  }

  renderActionBar() {
    const { backgroundColor } = this.context;
    const { twitterText, socialNetworkSelected } = this.state;
    const { isFacebookAvailable, isTwitterAvailable } = this.props;
  
    const backgroundColorStyle = { backgroundColor };

    const characterCounter = (socialNetworkSelected === 'twitter')
      ? <CharacterCounter currentCharacters={twitterText.length} maxCharacters={280} />
      : null;

    const postSwitcher = (isFacebookAvailable && isTwitterAvailable)
    ? <PostSwitcher socialNetworkSelected={socialNetworkSelected} toggleNetworkSelected={this.toggleNetworkSelected} />
    : null;

    const selectedNetworkPostBarStyles = {
      justifyContent: (socialNetworkSelected === 'twitter') ? 'space-between' : 'flex-end',
    };

    return (
      <View style={[styles.postOptionsBar, backgroundColorStyle, selectedNetworkPostBarStyles]}>
        {characterCounter}
        {postSwitcher}
      </View>
    );
  }

  renderPostBar() {
    const { backgroundColor } = this.context;
    const backgroundColorStyle = { backgroundColor };

    return (
      <View style={[styles.postBar, backgroundColorStyle]}>
        <CloseButton style={styles.closeButton} onPress={this.closeModal} isForModal />
        {this.renderWritePostLabel()}
        {this.renderPostButton()}
      </View>
    );
  }

  renderWritePostLabel() {
    const { screenTitleFacebook, screenTitleTwitter } = this.props;
    const { mainColor } = this.context;
    const textColorStyle = { color: mainColor };
    const { socialNetworkSelected } = this.state;

    const writeAPostLabel = (socialNetworkSelected === 'twitter')
      ? screenTitleTwitter
      : screenTitleFacebook;

    return <Text style={[styles.writePostLabel, textColorStyle]}>{writeAPostLabel}</Text>;
  }

  renderPostButton() {
    const { socialNetworkSelected } = this.state;
    const { postButtonText, tweetButtonText } = this.props;
    const { textColor } = this.context;
    const textColorStyle = { color: textColor };

    const postButtonLabel = (socialNetworkSelected === 'twitter')
      ? tweetButtonText
      : postButtonText;

    return (
      <TouchableOpacity style={styles.postButton} onPress={this.onPostPress}>
        <Text style={[styles.postButtonLabel, textColorStyle]}>{postButtonLabel}</Text>
      </TouchableOpacity>
    );
  }

  render() {
    const { postPlaceholderText } = this.props;
    const { backgroundColor } = this.context;
    const { keyboardHeight, facebookText, twitterText, socialNetworkSelected } = this.state;
    const text = (socialNetworkSelected === 'facebook') ? facebookText : twitterText;

    const screenBackgroundColor = { backgroundColor };
    const androidOffset = (Platform.OS !== 'ios') ? STATUS_BAR_HEIGHT : 0;
    const textInputHeight = (keyboardHeight)
    ? window.height - (2 * BAR_HEIGHT) - STATUS_BAR_HEIGHT - keyboardHeight - androidOffset
    : window.height - (2 * BAR_HEIGHT) - STATUS_BAR_HEIGHT - androidOffset;

    const textInputHeightStyle = { height: textInputHeight };

    return (
      <View style={[styles.writePostScreen, screenBackgroundColor]}>
        {this.renderPostBar()}
        <TextInput
          autoFocus
          multiline
          style={[styles.input, textInputHeightStyle]}
          placeholder={postPlaceholderText}
          placeholderTextColor={'#BBBAC1'}
          selectionColor={'#3350EE'}
          onChangeText={this.onTextChange}
          value={text}
        />
        {this.renderActionBar()}
      </View>
    );
  }
}

WritePostScreen.propTypes = {
  okButtonText: PropTypes.string,
  facebookErrorTitle: PropTypes.string,
  facebookErrorMessage: PropTypes.string,
  twitterErrorTitle: PropTypes.string,
  twitterErrorMessage: PropTypes.string,
  postPlaceholderText: PropTypes.string,
  postButtonText: PropTypes.string,
  tweetButtonText: PropTypes.string,
  screenTitleTwitter: PropTypes.string,
  screenTitleFacebook: PropTypes.string,
  facebookPageId: PropTypes.string,
  twitterScreenName: PropTypes.string,
  toggleModal: PropTypes.func,
  isFacebookAvailable: PropTypes.bool,
  isTwitterAvailable: PropTypes.bool,
};

WritePostScreen.contextTypes = {
  mainColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  textColor: PropTypes.string,
};

export default WritePostScreen;
