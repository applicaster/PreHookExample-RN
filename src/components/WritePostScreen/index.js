import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleModal } from '../../actions';

import WritePostScreen from './WritePostScreen';

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(WritePostScreen);
