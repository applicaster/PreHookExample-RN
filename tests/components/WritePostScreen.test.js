import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WritePostScreen from '../../src/components/WritePostScreen/WritePostScreen';

jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');
jest.mock('react-native-zapp-bridge', () => 'NativeZappBridge');

describe('WritePostScreen', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <WritePostScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly when only twitter is available', () => {
    const tree = renderer.create(
      <WritePostScreen isTwitterAvailable isFacebookAvailable={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('component renders correctly when only facebook is available', () => {
    const tree = renderer.create(
      <WritePostScreen isTwitterAvailable={false} isFacebookAvailable />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
