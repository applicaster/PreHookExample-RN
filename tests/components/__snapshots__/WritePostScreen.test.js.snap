// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritePostScreen component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#323232",
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 65,
          "justifyContent": "center",
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Image
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "uri": "feed_rn_close_button",
        }
      }
      style={
        Array [
          Object {
            "height": 25,
            "marginLeft": 5,
            "tintColor": "#FFFFFF",
            "width": 25,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": 30,
            "zIndex": 5,
          },
          Object {
            "tintColor": "#FFFFFF99",
          },
        ]
      }
      testID={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
            "top": 32,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
      Write a Post
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "position": "absolute",
          "right": 12,
          "top": 31,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "600",
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        Post
      </Text>
    </View>
  </View>
  <TextInput
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Write a post..."
    placeholderTextColor="#BBBAC1"
    selectionColor="#3350EE"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#000000",
          "fontSize": 20,
          "paddingHorizontal": 10,
          "paddingTop": 15,
          "textAlignVertical": "top",
          "width": 750,
        },
        Object {
          "height": 1224,
        },
      ]
    }
    value=""
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
        Object {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flexDirection": "column",
            "height": 22,
            "justifyContent": "center",
            "marginLeft": 11,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 10,
              "flexDirection": "column",
              "height": 22,
              "justifyContent": "center",
              "paddingHorizontal": 5,
            },
            Object {
              "backgroundColor": "#FFFFFF46",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          0/280
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`WritePostScreen component renders correctly when only facebook is available 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#323232",
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 65,
          "justifyContent": "center",
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Image
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "uri": "feed_rn_close_button",
        }
      }
      style={
        Array [
          Object {
            "height": 25,
            "marginLeft": 5,
            "tintColor": "#FFFFFF",
            "width": 25,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": 30,
            "zIndex": 5,
          },
          Object {
            "tintColor": "#FFFFFF99",
          },
        ]
      }
      testID={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
            "top": 32,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
      Write a Post
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "position": "absolute",
          "right": 12,
          "top": 31,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "600",
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        Post
      </Text>
    </View>
  </View>
  <TextInput
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Write a post..."
    placeholderTextColor="#BBBAC1"
    selectionColor="#3350EE"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#000000",
          "fontSize": 20,
          "paddingHorizontal": 10,
          "paddingTop": 15,
          "textAlignVertical": "top",
          "width": 750,
        },
        Object {
          "height": 1224,
        },
      ]
    }
    value=""
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
        Object {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flexDirection": "column",
            "height": 22,
            "justifyContent": "center",
            "marginLeft": 11,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 10,
              "flexDirection": "column",
              "height": 22,
              "justifyContent": "center",
              "paddingHorizontal": 5,
            },
            Object {
              "backgroundColor": "#FFFFFF46",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          0/280
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`WritePostScreen component renders correctly when only twitter is available 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#323232",
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 65,
          "justifyContent": "center",
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Image
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "uri": "feed_rn_close_button",
        }
      }
      style={
        Array [
          Object {
            "height": 25,
            "marginLeft": 5,
            "tintColor": "#FFFFFF",
            "width": 25,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": 30,
            "zIndex": 5,
          },
          Object {
            "tintColor": "#FFFFFF99",
          },
        ]
      }
      testID={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
            "top": 32,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    >
      Post a Tweet
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "position": "absolute",
          "right": 12,
          "top": 31,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "600",
            },
            Object {
              "color": undefined,
            },
          ]
        }
      >
        Tweet
      </Text>
    </View>
  </View>
  <TextInput
    autoFocus={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="Write a post..."
    placeholderTextColor="#BBBAC1"
    selectionColor="#3350EE"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "color": "#000000",
          "fontSize": 20,
          "paddingHorizontal": 10,
          "paddingTop": 15,
          "textAlignVertical": "top",
          "width": 750,
        },
        Object {
          "height": 1224,
        },
      ]
    }
    value=""
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 45,
          "width": 750,
        },
        Object {
          "backgroundColor": undefined,
        },
        Object {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "flexDirection": "column",
            "height": 22,
            "justifyContent": "center",
            "marginLeft": 11,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 10,
              "flexDirection": "column",
              "height": 22,
              "justifyContent": "center",
              "paddingHorizontal": 5,
            },
            Object {
              "backgroundColor": "#FFFFFF46",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 10,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          0/280
        </Text>
      </View>
    </View>
  </View>
</View>
`;
